### PR TITLE
#459 Fix empty rows being displayed incorrectly

### PR DESCRIPTION
**Pull request by Mathriok**
Empty lines should be displayed correctly with props.lines going through a function before being assigned as itemData. Try it on your machine!

### CODE REVIEW

**Comment by MelindaSW** [comment]
image

Looks so much better! It seems to work fine for me. I just had a thought, what if the replaceEmptyLinesWithHiddenChar function were to be placed in the engine instead? And the call to the function happened on row 209 for example, before sending the result to the frontend. Then the state would already contain the hidden chars and all the GUI would need to do was to handle the presentation.

**Comment by Mathriok** [comment]
Good idea! This next commit has the function moved to the engine.

File: src/js/engine/engine.js [hunk 1]
```diff
@@ -68,7 +74,7 @@ const openFile = async (sender, filePath) => {
       filePath,
       fileSize,
       endIndex,
-      history,
+      replaceEmptyLinesWithHiddenChar(history),
```
Comment (gustavjorlov) [inline src/js/engine/engine.js]: Some time in the future we might have forgotten why this function call exists. (The actual problem appears to be a frontend problem solved in the backend). Maybe just add a comment why the rendering needs something else than empty string to work?

**Comment by Mathriok** [comment]
Thanks for the feedback! Ignore the previous commit, I misthought a bit there. 
The replaceEmptyLines-function in the latest commit should now be able to handle empty spaces and empty lines containing empty space characters. Try it! Does anyone have a better suggestion for regular expressions to put in regList?

File: src/js/engine/engine.js [hunk 2]
```diff
@@ -54,6 +54,17 @@ const sendFileOpened = async (
   sender.send(ipcChannel, action);
 };
 
+// Invisible character U+2800 being used in line.replace
+const replaceEmptyLinesWithHiddenChar = arr => {
+  const regexList = [/^\s*$/];
+  return arr.map(line => {
+    const isMatch = regexList.some(rx => {
+      rx.test(line);
```
Comment (MelindaSW) [inline src/js/engine/engine.js]: I tried it with both rows containing "" and " ". And it works in my logviewer.
It did not work at first though, but after adding a return on row 62 no errors occured. 
`return rx.test(line)`

**Pull request merged**